tasmota/support.ino

```diff
@@ -1525,6 +1525,14 @@ void ModuleDefault(uint32_t module)
 void SetModuleType(void)
 {
   TasmotaGlobal.module_type = (USER_MODULE == Settings->module) ? Settings->user_template_base : Settings->module;
+
+  TasmotaGlobal.use_pwm_dimmer = false;
+#ifdef ESP8266
+  if (PWM_DIMMER == TasmotaGlobal.module_type) TasmotaGlobal.use_pwm_dimmer = true;
+#endif  // ESP8266
+#ifdef USE_LINKIND
+  if (LINKIND == TasmotaGlobal.module_type) TasmotaGlobal.use_pwm_dimmer = true;
+#endif  // USE_LINKIND
 }
```

Comment:
This won't work as module_types will be duplicated between esp8266 and esp32. I strongly suggest NOT to use the module_type for ESP32 specific code.

Your overall implemntation is much too invasive. There must be a better way to implement this specific hardware.

A possible solution is the way I implemented the Sonoff SPM hardware; it only uses one file and acts based on it's template name. The most preferred way is/was select functionality based on GPIO selection but for a POC using the template name is a much cleaner approach as it doesn't need small changes to many files.